### PR TITLE
Drop UnboundProjectFinder

### PR DESCRIPTION
**Pull request by duncanp-sonar**
- new CM won't require project-level changes

Part of #4120

### CODE REVIEW

File: src/Integration/Binding/BindingProcessImpl.cs [hunk 1]
```diff
@@ -321,37 +318,13 @@ private void InformAboutFilteredOutProjects()
...
-                    logger.WriteLine($"  {projectList}");
-                }
-                else
-                {
-                    logger.WriteLine(Strings.Bind_Ruleset_AllProjectsNeedToBeUpdated);
-                }
-            }
-            return projectsToUpdate;
+            // We no longer use rulesets => no projects need ruleset binding
+            return Array.Empty<Project>();
```
Comment (duncanp-sonar) [inline src/Integration/Binding/BindingProcessImpl.cs:327]: `BindingProcessImpl` is a complicated class. We might remove it entirely in the near future.
For now, I've settled for making the minimum changes to keep it compiling and the tests passing.

File: src/Integration/Binding/BindingProcessImpl.cs [hunk 2]
```diff
@@ -321,37 +318,13 @@ private void InformAboutFilteredOutProjects()
...
-            }
-            else
-            {
-                var unboundProjects = unboundProjectFinder.GetUnboundProjects()?.ToArray() ?? new Project[] { };
-                projectsToUpdate = projectsToUpdate.Intersect(unboundProjects).ToArray();
-                var upToDateProjects = allSupportedProjects.Except(unboundProjects);
-                if (upToDateProjects.Any())
-                {
-                    logger.WriteLine(Strings.Bind_Ruleset_SomeProjectsDoNotNeedToBeUpdated);
-                    var projectList = string.Join(", ", upToDateProjects.Select(p => p.Name));
```
Comment (duncanp-sonar) [inline src/Integration/Binding/BindingProcessImpl.cs:346]: I have a separate task to clean up unused resources once all of the unnecessary code has been deleted. However, in future PRs I'll try to do some cleanup as I go along.

File: src/Integration/LocalServices/IBindingChecker.cs [hunk 3]
```diff
@@ -58,15 +54,7 @@ public async Task<bool> IsBindingUpdateRequired(CancellationToken token)
...
-            var unboundProjects = unboundProjectFinder.GetUnboundProjects().ToArray();
-            var hasUnboundProjects = unboundProjects.Length > 0;
-
-            if (hasUnboundProjects)
-            {
-                logger.WriteLine(Strings.SonarLintFoundUnboundProjects, unboundProjects.Length, string.Join(", ", unboundProjects.Select(p => p.UniqueName)));
-            }
-
-            return hasUnboundProjects;
+            return false;
```
Comment (duncanp-sonar) [inline src/Integration/LocalServices/IBindingChecker.cs:57]: Minimal change to keep tests passing.
I suspect this whole class will be deleted later on.

**Review by bigfluffycookie — APPROVED** [review]
Very good!

**Pull request merged**